ibis/pyspark/compiler.py

```diff
@@ -98,19 +98,23 @@ def compile_selection(t, expr, scope, **kwargs):
     op = expr.op()
 
     src_table = t.translate(op.table, scope)
-    col_names_in_selection_order = []
 
+    col_in_selection_order = []
     for selection in op.selections:
         if isinstance(selection, types.TableExpr):
-            col_names_in_selection_order.extend(selection.columns)
+            col_in_selection_order.extend(selection.columns)
         elif isinstance(selection, (types.ColumnExpr, types.ScalarExpr)):
-            column_name = selection.get_name()
-            col_names_in_selection_order.append(column_name)
-            column = t.translate(selection, scope=scope)
-            src_table = src_table.withColumn(column_name, column)
+            col = t.translate(selection, scope=scope).alias(
+                selection.get_name()
+            )
+            col_in_selection_order.append(col)
+        else:
+            raise NotImplementedError(
```

Comment:
what hits this? can you test for it?

Reply:
Hmm..Nothing really hits this, I am just being cautious here. Maybe we should throw AssertionError instead?

Reply:
I can also remove the `else` if you prefer.

Reply:
neh this is ok